src/filter_finder.py

```diff
@@ -0,0 +1,208 @@
+								('C', 2, -1, 1)]
+
+		self.upper_controls = [	('g', 0, 1, 0),
+								('h', 1, 1, 0),
+								('j', 2, 1, 0),
+								('b', 0, -1, 0),
+								('n', 1, -1, 0),
+								('m', 2, -1, 0),
+
+								('G', 0, 1, 1),
+								('H', 1, 1, 1),
+								('J', 2, 1, 1),
+								('B', 0, -1, 1),
+								('N', 1, -1, 1),
+								('M', 2, -1, 1)]
+		
+		# initialize filters 
+		self.filters = {}
+		if unknown_filters is not None:
+			self.filters = self.get_filter_map(unknown_filters)
+		if known_filters is not None:
+			self.set_filters(known_filters)
+		# if both of the above conditions failed, use generic filter names
+		if self.filters == {}:
+			self.filters = self.get_filter_map("filter1", "filter2")
+
+	# display video at path and allow user to alter chosen color range live
+	def find_filter(self, path, chosen, downsize_scale = 2, write = True):
+		# if path leads to valid file
+		if os.path.isfile("../videos/Board_trim1.mp4"):
+			# if desired parameter exists in filtering parameter set
+			if chosen in self.filters:
+				# decide which color filter bounds to alter using controls based on supplied name
+				lower_chosen, upper_chosen = self.filters[chosen][0], self.filters[chosen][1]
+				
+				# try to read file
+				cap = cv2.VideoCapture(path)
+				success, frame = cap.read()
+				
+				if success:
+					# get dimensions of this video
+					height, width, _ = frame.shape
+					scaled_height, scaled_width = height/downsize_scale, width/downsize_scale
+
+					# parameters for displaying filter parameters on extracted frames
+					font = cv2.FONT_HERSHEY_SIMPLEX
+					font_position = (50,scaled_height - 40)
+					font_scale = .5
+					font_color = (0,0,255)
+					font_thickness = 2
+					
+					# dictionary of all extracted features to be displayed during runtime
+					features = {}
+					# dictionary of all binary masks used to extract those features
+					masks = {}
+					
+					# go through all the frames of this video, displaying extracted features
+					while success:
+						# rescale frame for faster processing
+						frame = cv2.resize(frame, (scaled_width, scaled_height))
+						hsv_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2HSV)
+						
+						features["raw"] = frame
+						
+						# get all masks and features for each filter
+						for key, (lower,upper) in self.filters.iteritems():
+							# create mask
+							masks[key] = cv2.inRange(hsv_frame, lower, upper)
+							# extract feature using mask
+							feature = cv2.bitwise_and(frame, frame, mask = masks[key])
+							
+							# write current filter info onto each filtered feature
+							lower_text = "lower: [{0},{1},{2}]".format(lower[0], lower[1], lower[2])
+							upper_text = "upper: [{0},{1},{2}]".format(upper[0],upper[1],upper[2])
+							cv2.putText(feature, lower_text, font_position, font, font_scale, font_color, font_thickness)
+							cv2.putText(feature, upper_text, (font_position[0], font_position[1]-30), font, font_scale, font_color, font_thickness)
```

Comment:
at this point you are getting to that "pyramid of doom thing", break into multiple methods.